packages/@okta/vuepress-site/guides/sign-into-spa/section03/react/attachcomponentsetc.md

```diff
@@ -1,18 +1,112 @@
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+// import config from ...
+
+class App extends Component {
+  render() {
+    return (
+      <Router>
+        <Security {...config} >
+          { /* child components have access to auth context */ }
+        </Security>
+      </Router>
+    );
+  }
+}
+
+export default App;
+```
 
 ## Provide the Login and Logout Buttons
 
-In the relevant location in your application, you want to provide `Login` and `Logout` buttons for the user. You can show/hide the correct button by using the `auth.isAuthenticated()` method. For example:
+In the relevant location in your application, you want to provide `Login` and `Logout` buttons for the user. You can show/hide the correct button by using the `auth.isAuthenticated()` method.  In this example,`props.auth` has been provided via the `withAuth` HOC.
+
+```javascript
+
+import { withAuth } from '@okta/okta-react';
+import React, { Component } from 'react';
+
+async function checkAuthentication() {
+  const authenticated = await this.props.auth.isAuthenticated();
+  if (authenticated !== this.state.authenticated) {
+    if (authenticated && !this.state.userinfo) {
+      const userinfo = await this.props.auth.getUser();
+      this.setState({ authenticated, userinfo });
+    } else {
+      this.setState({ authenticated });
+    }
+  }
+}
+
+export default withAuth(class Home extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { authenticated: null, userinfo: null };
+    this.checkAuthentication = checkAuthentication.bind(this);
+    this.login = this.login.bind(this);
+  }
+
+  async componentDidMount() {
+    this.checkAuthentication();
+  }
+
+  async componentDidUpdate() {
+    this.checkAuthentication();
+  }
+
+  async login() {
+    this.props.auth.login('/');
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.authenticated !== null &&
+        <div>
+          {this.state.authenticated &&
+            <div>
+              <p>Welcome back, {this.state.userinfo.name}!</p>
+            </div>
+          }
+          {!this.state.authenticated &&
+            <div>
+              <a onClick={this.login}>Login</a>
+            </div>
+          }
+        </div>
+        }
+      </div>
+    );
+  }
+});
+```
+
+## Create the Callback Handler
+
+In order to handle the redirect back from Okta, you need to capture the token values from the callback URL `/implicit/callback` and pass them to the handleAuthentication() method of the Auth object. We have provided `ImplicitCallback` which implements this logic. It can be mapped to a route as shown:
 
-{code}
 
-## Update Your App.js
+```javascript
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { Security, ImplicitCallback } from '@okta/okta-react';
+// import config from ...
 
-Finally, pass in your configuration into `Security` and connect your application's paths:
```

Comment:
is this not an appropriate explanation sentence for the code? Maybe I'm not following what is happening here, and the sentence isn't needed? Providing a bit of guidance for each step along the way is helpful...but maybe not necessary here and I'm just not seeing it?